contracts/managers/triggers/MovingAverageToAssetPriceCrossoverTrigger.sol

```diff
@@ -0,0 +1,239 @@
+     * from last confirmed state then timestamp is logged to be used in calculating the start
+     * and end of the confirmation period.
+     */
+    function initialTrigger()
+        external
+    {
+        // Make sure enough time has elapsed since last initialTrigger
+        require(
+            hasConfirmationWindowElapsed(),
+            "MovingAverageToAssetPriceCrossoverTrigger.initialTrigger: Not enough time passed from last initial crossover."
+        );
+
+        // Get current market state and check that it's different from last confirmed state
+        bool currentMarketState = getCurrentMarketState();
+        require(
+            currentMarketState != lastConfirmedState,
+            "MovingAverageToAssetPriceCrossoverTrigger.initialTrigger: Market conditions have not changed since last confirmed state."
+        );
+
+        lastInitialTriggerTimestamp = block.timestamp;
+    }
+
+    /*
+     * If within the confirmation time period, the current market state is calculated then
+     * compared to the last confirmed market state. If current state differs from last confirmed
+     * state then the last confirmed state is updated to the current market state.
+     */
+    function confirmTrigger()
+        external
+    {
+        // Make sure currently in confirmation window
+        require(
+            inConfirmationWindow(),
+            "MovingAverageToAssetPriceCrossoverTrigger.confirmPropose: Confirming signal must be within bounds of the confirm propose."
+        );
+
+        // Get current market state and check that it's different from last confirmed state
+        bool currentMarketState = getCurrentMarketState();
+        require(
+            currentMarketState != lastConfirmedState,
+            "MovingAverageToAssetPriceCrossoverTrigger.confirmTrigger: Market conditions have not changed since last confirmed state."
+        );
+
+        lastConfirmedState = currentMarketState;
+        triggerFlippedIndex = triggerFlippedIndex.add(1);
+
+        emit TriggerFlipped(currentMarketState, triggerFlippedIndex, block.timestamp);
+    }
+
+    /*
+     * Returns if trigger is in bullish state.
+     *
+     * @return             Whether market conditions are bullish
+     */
+    function isBullish()
+        external
+        view
+        returns (bool)
+    {
+        return lastConfirmedState;
+    }
+
+    /*
+     * Return if initialTrigger can be called without reverting.
+     *
+     * @return       Whether in can initialTrigger
+     */
+    function canInitialTrigger()
+        external
+        view
+        returns (bool)
+    {
+        if (hasConfirmationWindowElapsed()) {
+            return getCurrentMarketState() != lastConfirmedState;
+        } else {
+            return false;
+        }
+    }
+
+    /*
+     * Return if confirmTrigger can be called without reverting.
+     *
+     * @return       Whether in can confirmTrigger
+     */
+    function canConfirmTrigger()
```

Comment:
Since we are thinking about it, let's consider adding the automatic confirmation if the price goes beyond the MA by a certain percentage.